filesystem/src/main/java/org/frankframework/filesystem/exchange/ExchangeFileSystem.java

```diff
@@ -0,0 +1,582 @@
+	private MsalClientAdapter msalClientAdapter;
+	private GraphClient client;
+	private MailFolder mailFolder;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		if (StringUtils.isEmpty(getUrl()) && StringUtils.isEmpty(getMailAddress())) {
+			throw new ConfigurationException("either url or mailAddress needs to be specified");
+		}
+
+		credentials = new CredentialFactory(getAuthAlias(), getClientId(), getClientSecret());
+		if (StringUtils.isNotEmpty(getProxyHost()) && (StringUtils.isNotEmpty(getProxyAuthAlias()) || StringUtils.isNotEmpty(getProxyUsername()) || StringUtils.isNotEmpty(getProxyPassword()))) {
+			proxyCredentials = new CredentialFactory(getProxyAuthAlias(), getProxyUsername(), getProxyPassword());
+		}
+
+		msalClientAdapter = SpringUtils.createBean(applicationContext, MsalClientAdapter.class);
+		msalClientAdapter.setProxyHost(getProxyHost());
+		msalClientAdapter.setProxyPort(getProxyPort());
+		CredentialFactory proxyCf = getProxyCredentials();
+		if (proxyCf != null) {
+			msalClientAdapter.setProxyUsername(proxyCf.getUsername());
+			msalClientAdapter.setProxyPassword(proxyCf.getPassword());
+		}
+
+		msalClientAdapter.setKeystore(getKeystore());
+		msalClientAdapter.setKeystoreType(getKeystoreType());
+		msalClientAdapter.setKeystoreAuthAlias(getKeystoreAuthAlias());
+		msalClientAdapter.setKeystorePassword(getKeystorePassword());
+		msalClientAdapter.setKeystoreAlias(getKeystoreAlias());
+		msalClientAdapter.setKeystoreAliasAuthAlias(getKeystoreAliasAuthAlias());
+		msalClientAdapter.setKeystoreAliasPassword(getKeystoreAliasPassword());
+		msalClientAdapter.setKeyManagerAlgorithm(getKeyManagerAlgorithm());
+
+		msalClientAdapter.setTruststore(getTruststore());
+		msalClientAdapter.setTruststoreType(getTruststoreType());
+		msalClientAdapter.setTruststoreAuthAlias(getTruststoreAuthAlias());
+		msalClientAdapter.setTruststorePassword(getTruststorePassword());
+		msalClientAdapter.setTrustManagerAlgorithm(getTrustManagerAlgorithm());
+		msalClientAdapter.setVerifyHostname(isVerifyHostname());
+		msalClientAdapter.setAllowSelfSignedCertificates(isAllowSelfSignedCertificates());
+		msalClientAdapter.setIgnoreCertificateExpiredException(isIgnoreCertificateExpiredException());
+
+		msalClientAdapter.configure();
+
+		if (StringUtils.isBlank(baseFolder)) {
+			baseFolder = "Inbox";
+		}
+	}
+
+	@Override
+	public void open() {
+		try {
+			super.open();
+
+			msalClientAdapter.start();
+			client = msalClientAdapter.createGraphClient(tenantId, getCredentials());
+
+			List<String> folder = StringUtil.split(baseFolder, "/");
+
+			String rootFolder = folder.remove(0);
+			List<MailFolder> folders = client.getMailFolders(mailAddress);
+			MailFolder foundMailFolder = folders.stream()
+					.filter(t -> rootFolder.equalsIgnoreCase(t.getName()))
+					.findFirst()
+					.orElseThrow(() -> {
+				throw new LifecycleException("unable to find folder [%s] in mailbox [%s]".formatted(rootFolder, mailAddress));
+			});
```

Comment:
Je haalt hiermee max 20 folders op. Als een mailbox meer dan 20 folders heeft, en de folder die je zoekt zit niet in de eerste 20, dan vind je die dus nooit en kan je nooit het FS openen -- ondanks dat de folder eigenlijk wel bestaat.

Wil je aan dit probleem nog iets doen in dit PR? Of in een volgend PR?

Reply:
20 per request, en 20 requests :)

Reply:
In de `orElseThrow` hoor je trouwens niet zelf te throwen, alleen de `new Exception` terug te geven.